core/src/main/java/org/elasticsearch/plugins/InstallPluginCommand.java

```diff
@@ -301,6 +296,22 @@ private Path unzip(Path zip, Path pluginsDir) throws IOException, UserError {
         return target;
     }
 
+    private Path stagingDirectory(Path pluginsDir) throws IOException {
+        try {
+            Set<PosixFilePermission> perms = new HashSet<>();
+            perms.add(PosixFilePermission.OWNER_EXECUTE);
+            perms.add(PosixFilePermission.OWNER_READ);
+            perms.add(PosixFilePermission.OWNER_WRITE);
+            perms.add(PosixFilePermission.GROUP_READ);
+            perms.add(PosixFilePermission.GROUP_EXECUTE);
+            perms.add(PosixFilePermission.OTHERS_READ);
+            perms.add(PosixFilePermission.OTHERS_EXECUTE);
+            return Files.createTempDirectory(pluginsDir, ".installing-", PosixFilePermissions.asFileAttribute(perms));
+        } catch (IllegalArgumentException | UnsupportedOperationException e) {
+            return Files.createTempDirectory(pluginsDir, ".installing-");
```

Comment:
You can just change the throws clause for this exception. All the cli tools execute methods throw Exception, so actual errors can propagate to java main instead of adding wrapped exceptions everywhere.